doc: Document the Windows CI job

petrutlucian94 opened the pull request:
We're adding a document that describes:

* the steps performed by the Windows CI job
* artifact structure
* frequently asked questions

The document is meant to assist the Ceph developers in investigating CI failures. This is especially important as the Windows CI job runs integration tests that would otherwise only be executed by Teuthology, thus helping catch potential regressions quickly.

Note that the identified regressions are not necessarily Windows specific, usually affecting Linux builds as well.





<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

zdover23 commented on doc/dev/developer_guide/tests-windows.rst:
```diff
@@ -0,0 +1,126 @@
+.. image:: ../../images/windows_ci_status_page.png
+      :align: center
+
+The Windows CI artifacts can be downloaded as a zip archive or viwed inside
```
viewed

zdover23 commented on doc/dev/developer_guide/tests-windows.rst:
```diff
@@ -0,0 +1,126 @@
+.. image:: ../../images/windows_ci_artifacts.png
+      :align: center
+
+Artiact contents:
```
Artifact

petrutlucian94 replied:
Thanks. Sorry for the typos, wrote this in a hurry.

zdover23 replied:
No, no. These are notes to self. I'll correct all this stuff in a future PR.

I think this is a great contribution. I'm not meaning to be nit-picky.

Also, you need to add a Signed-off-by: line.

zdover23 replied:
Signed-off-by: line instructions can be read here:

https://docs.ceph.com/en/reef/dev/developer_guide/basic-workflow/#configuring-your-local-environment

zdover23 replied:
Actually, you just need to put "doc:" at the beginning of the title of this PR.

zdover23 commented on doc/dev/developer_guide/tests-windows.rst:
```diff
@@ -0,0 +1,126 @@
+Lab infrastructure. These tests are triggered on-demand by the Ceph QA
+team and do not run automatically for every submitted pull request.
+
+Since the Windows CI job only focuses on the client-side Ceph components,
```
focuses only

zdover23 commented on doc/dev/developer_guide/tests-windows.rst:
```diff
@@ -0,0 +1,126 @@
+
+For this reason, the Windows CI often catches regressions that are missed by the
+other checks and would otherwise only come up through Teuthology. More often
+than not, these regressions are not platform specific and affect Linux as well.
```
platform-specific

zdover23 reviewed (approved):
This is a great contribution.

I'll merge this as soon as the tests are passed and I'll backport it.

There are minor grammar and comma-splice complaints, but nothing that I can't address in a future PR.

Thanks for this.

petrutlucian94 commented:
@zdover23 thank you so much for reviewing this on such a short notice. I've addressed the issues and added a link on the Windows install page for better visibility.

zdover23 commented:
> @zdover23 thank you so much for reviewing this on such a short notice. I've addressed the issues and added a link on the Windows install page for better visibility.

I'm glad to help. I just happened to be at the terminal, working on a separate matter and waiting for a build to complete. We got lucky.

petrutlucian94 commented:
Oh, forgot to add an important note:

```
Timeouts and missing test results are often an indication that a process crashed. Note that the ceph status
is printed out on the console before and after performing the tests, which can help identify crashed services.

You may also want to check the service logs (both client and server side). Also, be aware that the Windows
"application" event log will contain entries in case of crashed Windows processes.
```

idryomov commented on doc/dev/developer_guide/tests-windows.rst:
```diff
@@ -0,0 +1,126 @@
+on GitHub. **In other words, it runs various librados, librbd and libcephfs
+tests that other checks such as "make check" do not.**
+
+For this reason, the Windows CI often catches regressions that are missed by the
```
Nit: Might want to mention explicitly that the Windows job builds Ceph in Release rather than Debug mode and with a different set of `-D` options passed to CMake, disabling many of the things/components that are enabled in other CI checks.  This sometimes leads to FTBFS just for the Windows job due to missing ifdefs and such in the code.

petrutlucian94 replied:
Good point, I'd also mention the toolchain version as well as other dependencies such as boost.

zdover23 commented:
https://github.com/ceph/ceph/pull/60033 - Squid backport
https://github.com/ceph/ceph/pull/60034 - Reef backport
https://github.com/ceph/ceph/pull/60035 - Quincy backport
